src/SwapRouter.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {IUniversalRouter} from "src/interfaces/IUniversalRouter.sol";
+import {ICreatorToken} from "src/interfaces/ICreatorToken.sol";
+import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
+
+contract SwapRouter {
+  IUniversalRouter private constant UNIVERSAL_ROUTER =
+    IUniversalRouter(0x198EF79F1F515F02dFE9e3115eD9fC07183f02fC); // Base
+
+  address public constant WETH_ADDRESS = 0x4200000000000000000000000000000000000006;
+  address public constant USDC_ADDRESS = 0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913;
+  bytes private path;
+  // for WRAP_ETH, V3_SWAP_EXACT_OUT, and UNWRAP_WETH
+  bytes private constant COMMANDS =
+    abi.encodePacked(bytes1(uint8(0x0b)), bytes1(uint8(0x01)), bytes1(uint8(0x0c)));
+  bytes3 private constant LOW_FEE_TIER = bytes3(uint24(500));
+
+  constructor() {
+    // WETH -> USDC, order is switched because of V3_SWAP_EXACT_OUT
+    path =
+      bytes.concat(bytes20(address(USDC_ADDRESS)), LOW_FEE_TIER, bytes20(address(WETH_ADDRESS)));
```

Comment:
I've set it outside the constructor in [ced1396](https://github.com/showtime-xyz/creator-tokens/pull/44/commits/ced1396d07f469ee1a09006a368a8c1ebedb91ac), but couldn't make it constant because of the `Initial value for constant variable has to be compile-time constant` error. I'm assuming it's because we're trying to use other constant variables.

Reply:
yeah the use of `bytes.concat` prevents it from being assigned as constant, so I guess I was recommending we pre-calculate it and hardcode it. You're right that if we end up passing in the other values, this doesn't work.

Reply:
Do we still want to pass in USDC and WETH addresses in the constructor even tho the `universalRouter` might not be working on base goerli?

Reply:
Yep. We should make that configurable and explore deploying on an alternate testnet.